### PR TITLE
Drop FieldValue from tests

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
- Drops the FieldValue test classes
- `wrap()` returns `Value` and replaces `valueOf()` test helper
- Moved `FieldValueTest.testValueEquality()`, `FieldValueTest.testValueOrdering()` and `FieldValueTest.testCanonicalIds()` to a new ProtoValuesTest files that test the ProtoValues functionality directly (via a small FieldValue-alike helper)

This is the last cleanup PR before I want to merge to master. We can discuss if we should proceed with the suggested renames as well.

### CODE REVIEW

File: firebase-firestore/src/test/java/com/google/firebase/firestore/model/ObjectValueBuilderTest.java [hunk 1]
```diff
@@ -31,9 +31,9 @@
 @Config(manifest = Config.NONE)
 public class ObjectValueBuilderTest {
   private String fooString = "foo";
-  private Value fooValue = valueOf(fooString);
+  private Value fooValue = TestUtil.wrap(fooString);
```
Comment (wilhuff) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/model/ObjectValueBuilderTest.java]: Static import `TestUtil.wrap` (to match other usages above).

File: firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java [hunk 2]
```diff
@@ -118,16 +116,15 @@ public void setUp() {
...
     assertEquals(proto, actual);
-    assertTrue(ProtoValues.equals(value.getProto(), proto));
+    assertTrue(ProtoValues.equals(actual, proto));
   }
 
   @Test
   public void testEncodesNull() {
-    FieldValue value = wrap(null);
+    Value value = wrap(null);
     com.google.firestore.v1.Value proto = valueBuilder().setNullValueValue(0).build();
```
Comment (wilhuff) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java]: Replace fully qualified `com.google.firestore.v1.Value` declarations in here with just `Value`?

File: firebase-firestore/src/test/java/com/google/firebase/firestore/model/ObjectValueBuilderTest.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/model/ObjectValueBuilderTest.java]: Done

File: firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java (hunk 2 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java]: Done

**Pull request merged**